nifi-nar-bundles/nifi-mqtt-bundle/nifi-mqtt-processors/src/main/java/org/apache/nifi/processors/mqtt/ConsumeMQTT.java

```diff
@@ -194,9 +194,15 @@ public void onPropertyModified(PropertyDescriptor descriptor, String oldValue, S
         }
 
         final boolean clientIDSet = context.getProperty(PROP_CLIENTID).isSet();
+        final boolean clientIDwithEL = context.getProperty(PROP_CLIENTID).isExpressionLanguagePresent();
         final boolean groupIDSet = context.getProperty(PROP_GROUPID).isSet();
-        if (clientIDSet && groupIDSet) {
-            results.add(new ValidationResult.Builder().subject("Client ID and Group ID").valid(false).explanation("if client ID is not unique, multiple nodes cannot join the consumer group").build());
+        if (!clientIDwithEL && clientIDSet && groupIDSet) {
+            results.add(new ValidationResult.Builder()
+                    .subject("Client ID and Group ID").valid(false)
+                    .explanation("If client ID is not unique, multiple nodes cannot join the consumer group. If you want "
```

Comment:
The explanation will be concatenated into a sentence like `"... is invalid because [explanation]"` by the framework.
So it should start with lower case I think.